### ISSUE TITLE
Can't edit layer thickness in IfcMaterialLayerSet for types or objects

### ISSUE THREAD

**Issue by Jumalvelho** [IfcOpenShell/IfcOpenShell#5850]
When clicking "Enable Editing Assigned Material" button  in "Object materials" when IfcMaterialLayerSet is use, I cannot find any prompt to edit the thickness of the layers anymore:

![image](https://github.com/user-attachments/assets/6955f611-83a8-4514-9c58-251218420dea)
![image](https://github.com/user-attachments/assets/39d12c37-1a8a-4a52-b6f8-249f057ece51)

Running Blender 4.3.0 and Bonsai 0,8.1-alpha241206-c4ffc06 on Win 10.

**Comment by sboddy** [comment]
Huh! That looks like exactly what I just came to report. Did you delete the material set and then try to add a new one? If so, it is exaclty what I did, and it locked me in to walls whose heights cannot be edited. I see the same half rendered UI in the material editor too.

For the devs here's the exception I see on the console:
```
Traceback (most recent call last):
  File "/home/steve/.config/blender/4.2bonsai/extensions/.local/lib/python3.11/site-packages/bonsai/bim/module/material/ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/steve/.config/blender/4.2bonsai/extensions/.local/lib/python3.11/site-packages/bonsai/bim/module/material/ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "/home/steve/.config/blender/4.2bonsai/extensions/.local/lib/python3.11/site-packages/bonsai/bim/module/material/ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/steve/.config/blender/4.2bonsai/extensions/.local/lib/python3.11/site-packages/bonsai/bim/module/material/ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'
```
Haven't looked at it myself yet, but it looks to be a fairly obviously trying to read a value before it gets set. Seems like it is missing setting the value when adding a new IfcMaterialLayerSet to an object having removed it from the Wall and the WallType. It's too late here to go looking into this, but I might look if I get chance.

@Jumalvelho It might help if you open the console (windows) or run from console (linux) and confirm if you are seeing the same error on the console.

**Comment by sboddy** [comment]
A really temporary dumb workaround is to open the source file in your blender config folder and wrap the line in a try/except clause. The daft thing is only checked to set an inconsequential text label, so it is not critical. At least then you can edit the MaterialLayerSet.
```
        try:
            layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
        except:
            layer_set_direction = 'AXIS2'
```
Ironically this did not fix my wall height issue, at least not straight away. I had to assign them another WallType, then set them back to the original WallType after I fixed the MaterialLayerSet. _Then_ I could edit the heights.
As I said... *Not* a proper fix. No warranty expressed or implied!

**Comment by Jumalvelho** [comment]
Okay, this seems to be the same error, this is my console:

```Traceback (most recent call last):
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'
Traceback (most recent call last):
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'
Traceback (most recent call last):
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'
Traceback (most recent call last):
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'
Traceback (most recent call last):
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'
Traceback (most recent call last):
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'
Traceback (most recent call last):
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'
Traceback (most recent call last):
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'
Traceback (most recent call last):
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'
Traceback (most recent call last):
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 173, in draw
    return self.draw_material_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 199, in draw_material_ui
    self.draw_set_ui()
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 216, in draw_set_ui
    return self.draw_editable_set_ui()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\[user]\AppData\Roaming\Blender Foundation\Blender\4.3\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\material\ui.py", line 237, in draw_editable_set_ui
    layer_set_direction = self.props.material_set_usage_attributes["LayerSetDirection"].enum_value
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
KeyError: 'bpy_prop_collection[key]: key "LayerSetDirection" not found'```

**Comment by Andrej730** [comment]
Occurred after b52edce for editing non-IfcMaterialLayerSetUsage materials, fixed in a9defb9.
